MRB-665 Support for global models and remove meteodata-lab

frazane opened the pull request:
This PR introduces minimal support for global models in evalml, as we might want to evaluate small experiments on coarse global grids (e.g. o48 or o96).

**Summary of changes**
* optionally disable setting the `ECCODES_DEFINITION_PATH` environment variable when running global models 
* added base anemoi-inference config for global models and an example evalml config
* switch from meteodatalab to earthkit-data (with its xarray engine) to load GRIB files during verification 

TODO:
- [ ] Deal with the field `_earthkit` added to the xarray object by earthkit. New versions of xarray crash when to_netcdf with fields that are not allowed. Solution is to always `pop` it away. Temporary solution is a pin of xarray but its not sustainable.

frazane committed:
read data with earthkit

frazane committed:
optionally disable eccodes local definitions

frazane committed:
add minimal configs

dnerini committed:
Merge branch 'main' into feat/support-global-models

dnerini committed:
Update config

dnerini committed:
Merge branch 'main' into feat/support-global-models

dnerini committed:
Merge branch 'main' into feat/support-global-models

dnerini committed:
Change example config to stage B checkpoint

dnerini committed:
Merge branch 'main' into feat/support-global-models

dnerini committed:
Merge branch 'main' into feat/support-global-models

dnerini committed:
Merge branch 'main' into feat/support-global-models

dnerini committed:
Stage B models work

dnerini committed:
Fix config

dnerini commented:
experiment workflows works, e.g
<img width="676" height="637" alt="image" src="https://github.com/user-attachments/assets/540aa735-92f2-407a-9258-5bf10a8fbc3e" />

I'm currently debugging the showcase workflow, but I suppose it'd make already sense to have another review round here.

dnerini committed:
Support for tp

dnerini commented:
@frazane can you also have a look pls? Supporting tp from the global GRIB files has been a bit of a pain and some of this fixes are really unpleasant ....

dnerini committed:
lint

dnerini commented on src/data_input/__init__.py:
```diff
@@ -108,9 +135,43 @@ def load_fct_data_from_grib(
+    params_sel = list(
+        {p for p in params} | {_COSMO_TO_IFS[p] for p in params if p in _COSMO_TO_IFS}
+    )
+    # Precipitation params don't have a step=0 field (accumulation is zero at
```
I suppose this was previously handled by meteodata-lab?

jonasbhend replied:
I guess this didn't occur previously because we haven't read from the global files at all. All our lam cutouts use COSMO names.

MicheleCattaneo commented:
Is it possible that there is an issue with the verification at specific locations?
<img width="640" height="480" alt="202402010000_T_2M_KLO" src="https://github.com/user-attachments/assets/3a5d984b-1332-41bb-871f-4f3376ae6b18" />

This is T_2M. I understand that ERA is 0.25 degrees, but there is not even the diurnal cycle here. What do you think? Another issue, the lines dont start from the same point, but this model uses ERA5 as IC (unless I have an issue in my evalml config?)

dnerini committed:
Merge branch 'main' into feat/support-global-models

dnerini committed:
Merge branch 'main' into feat/support-global-models

dnerini committed:
Linting

jonasbhend committed:
Add lat lon information to ICON baselines (#150)

These changes have not been reconciled with main. Therefore another PR
is necessary.

---------

Co-authored-by: Francesco Zanetta <62377868+frazane@users.noreply.github.com>
Co-authored-by: Claire Merker <34312518+clairemerker@users.noreply.github.com>

jonasbhend commented on src/data_input/__init__.py:
```diff
@@ -123,11 +138,28 @@ def load_fct_data_from_grib(
+        datasets.append(fieldlist.sel(param=other_params, step=steps).to_xarray(profile=profile))
+    if prec_params:
+        prec_steps = [s for s in steps if s > 0]
+        datasets.append(fieldlist.sel(param=prec_params, step=prec_steps).to_xarray(profile=profile))
```
in `data_extract_baseline.py` we use the following pattern:

```
       for lt in lead_times:
            lh = lt % 24
            ld = lt // 24
            filepath = file / "grib" / f"{gribname}{ld:02}{lh:02}0000_{run_id}"
            LOG.info(f"Extracting {filepath}.")
            fields = ekd.from_source("file", filepath)
            for field in fields:
                if field.metadata("shortName") in params:
                    out.append(field)
```
That one looks slow to me, but I haven't profiled.

jonasbhend replied:
The above approach has the advantage that you do not hard-code expectations in, in the sense that if there was a precipitation field at time 0 (with zero or NA in it), then you could still read it.

jonasbhend committed:
Blacklist of forecast initializations (#151)

When running experiments with many initialization times, sometime
individual initializations do not run successfully due to data problems.
In order to deal with this problem, a blacklist of initialization times
can be provided in the config to exclude from experiments.

## Summary of Changes
* adapted config schema and update generation of list of initialization
times
* provide one example config with a blacklisted initialization

MicheleCattaneo committed:
chore: temporary fix of dependencies. Before merging we need to deal with _earthkit field

MicheleCattaneo committed:
fix: removed hardcoded verification domain equal to the ICON domain, allowed to be the globe

frazane committed:
fix docstrings

frazane committed:
update dependencies

frazane committed:
fix config

cosunae commented:
DWD eccodes released on open data version 2.44 http://opendata.dwd.de/
we have a private copy of 2.47

MicheleCattaneo committed:
fix: temporary fix; removed map_forecast_to_truth as it was creating wrong results comparing o96 against o96

MicheleCattaneo committed:
chore: various fixes for eccodes and grib. Fixed env compitation

MicheleCattaneo committed:
chore: updated swissAI o96 config
